apps/wallet/src/features/Swap/InputSwap/FromToken.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMemo } from 'react';
+import { StyleSheet, TextInput } from 'react-native';
+import { Button, Text, View } from '@walless/gui';
+import { useSnapshot } from 'utils/hooks';
+
+import type { SwapContext } from '../context';
+import { swapActions, swapContext } from '../context';
+
+import SelectButton from './SelectButton';
+
+const FromToken = () => {
+	const { fromToken, amount } = useSnapshot(swapContext).swap as SwapContext;
+
+	const balance = useMemo(() => {
+		if (!fromToken) return 0;
+		return (
+			parseFloat(fromToken.account.balance) / 10 ** fromToken.account.decimals
```

Comment:
promote `10 ** fromToken.account.decimals` into a variable, to improve readability.